doc/source/serve/production.md

```diff
@@ -583,3 +565,217 @@ This section summarizes the best practices when deploying to production:
+# A head node service maintained by the RayCluster.
+rayservice-sample-raycluster-qd2vl-head-svc        ClusterIP   10.100.180.221   <none>        6379/TCP,8265/TCP,10001/TCP,8000/TCP,52365/TCP   24m
+# A serve service maintained by the RayService.
+rayservice-sample-serve-svc                        ClusterIP   10.100.39.92     <none>        8000/TCP                                         24m
+```
+
+### Querying the application
+
+Once the `RayService` is running, we can query it over HTTP using the service created by the controller.
+This service can be queried directly from inside the cluster, but to access it from your laptop you'll need to configure a Kubernetes ingress or use port forwarding as below:
+
+```console
+$ kubectl port-forward service/rayservice-sample-serve-svc 8000
+$ curl -X POST -H 'Content-Type: application/json' localhost:8000 -d '["MANGO", 2]'
+6
+```
+
+### Getting the status of the application
+
+As the `RayService` is running, the `KubeRay` controller continually monitors it and writes relevant status updates to the CR.
+You can view the status of the application using `kubectl describe`.
+This includes the status of the cluster, any events that happened such as health check failures or restarts, and the same application-level statuses reported by [`serve status`](serve-in-production-inspecting).
+
+```console
+$ kubectl get rayservices
+NAME                AGE
+rayservice-sample   7s
+
+$ kubectl describe rayservice rayservice-sample
+```
+
+### Updating the application
+
+To update the `RayService`, modify the manifest and apply it use `kubectl apply`.
+There are two types of updates that can occur:
+- *Application-level updates*: when only the Serve config options are changed, the update is applied _in-place_ on the same Ray cluster. This enables [lightweight updates](serve-in-production-lightweight-update) such as scaling a deployment up or down or modifying autoscaling parameters.
+- *Cluster-level updates*: when the `RayCluster` config options are changed, such as updating the container image for the cluster, it may result in a cluster-level update. In this case, a new cluster is started, the application is deployed to it, and once it is running the Kubernetes service is updated to point at the new cluster. There should not be any downtime for the application, but note that this requires the Kubernetes cluster to be large enough to schedule both Ray clusters.
+
+#### Example: Serve config update
+
+In the `FruitStand` example above, let's change the price of a mango in the Serve config to 4:
+
+```console
+  - name: MangoStand
+    numReplicas: 1
+    userConfig: |
+      price: 4
+```
+
+Now to update the application we apply the modified manifest:
+
+```console
+$ kubectl apply -f ray_v1alpha1_rayservice.yaml
+
+$ kubectl describe
+...
+  serveDeploymentStatuses:
+  - healthLastUpdateTime: "2022-07-18T21:51:37Z"
+    lastUpdateTime: "2022-07-18T21:51:41Z"
+    name: MangoStand
+    status: UPDATING
```

Comment:
Does the status actually change to `UPDATING`? After #27000, updating the `user_config` should be a lightweight config update, so the status should continuously be `HEALTHY`. Did any other `MangoStand` field get changed?

Reply:
even on a user_config change it should briefly get changed to `UPDATING`

Reply:
though realistically it'll probably be too fast to catch with the CLI. I think it's ok, mostly here for instructive purposes

Reply:
That sounds good.